bin/common.ml

```diff
@@ -799,12 +799,34 @@ let term =
             "Force actions associated to aliases to be re-executed even\n\
             \                   if their dependencies haven't changed.")
   and+ watch =
-    Arg.(
-      value & flag
-      & info [ "watch"; "w" ]
-          ~doc:
-            "Instead of terminating build after completion, wait continuously \
-             for file changes.")
+    Term.ret
+      (let watch_arg_name = "watch" in
+       let passive_watch_mode_arg_name = "passive-watch-mode" in
+       let+ w_flag =
+         Arg.(
+           value & flag
+           & info [ watch_arg_name; "w" ]
+               ~doc:
+                 "Instead of terminating build after completion, wait \
+                  continuously for file changes.")
+       and+ passive_watch_mode_flag =
+         Arg.(
+           value & flag
+           & info
+               [ passive_watch_mode_arg_name ]
+               ~doc:
+                 "Do not automatically start or re-start a build. Only do it \
+                  when a build RPC is received.")
+       in
+       match (w_flag, passive_watch_mode_flag) with
+       | false, false -> `Ok Dune_engine.Watch_mode_config.No
+       | false, true ->
+         `Error
+           ( true
+           , sprintf "Cannot use %s if %s was not passed"
+               passive_watch_mode_arg_name watch_arg_name )
+       | true, false -> `Ok (Dune_engine.Watch_mode_config.Yes Eager)
+       | true, true -> `Ok (Dune_engine.Watch_mode_config.Yes Passive))
```

Comment:
You could also make these two options mutually exclusive by using `one_of` (defined locally in `common.ml`). That would change a bit the CLI: you would have to pass `--pasive-watch-mode` rather than `--watch --passive-watch-mode`.

Another option is to use an optional parameter. i.e:
- nothing or `--watch=no` for no watch mode
- `--watch` or `--watch=eager` for eager watch mode
- `--watch=passive` for passive watch mode

Reply:
That sounds good. I like the second option a bit better, so I'll go with that.

Reply:
Pushed this change. To make the change handle `-wpfoo` backwards-compatibly, made a small tweak to our cmdliner fork.